fix: Allow map_docs to accommodate 'args' sent from client call

marination opened the pull request:
- 'args' can be sent  to the mapper method from the client call (eg. map_current_doc)
- map_docs didn't handle the same

**After fix:**
- Can optionally send args to mapper method.

marination committed:
fix: Modified map_docs to accomodate 'args' sent from client call

surajshetty3416 commented on frappe/model/mapper.py:
```diff
@@ -30,16 +30,18 @@ def make_mapped_doc(method, source_name, selected_children=None, args=None):
 @frappe.whitelist()
-def map_docs(method, source_names, target_doc):
+def map_docs(method, source_names, target_doc, args:str=None):
+	# args => "{ 'supplier': 100 }"
```
Better to make this a part of docstring instead of a comment.

marination committed:
fix: Comment to docstring and minor changes

- added description for args with example
- 'yield' might not be the best idea, barely any improvement, so changed back to 'return'
